### PR TITLE
remove unused flake-inputs

### PR DESCRIPTION
**Pull request by Mic92**
nixos-facter brings in a lot of flake inputs while users only want nixos modules that do not have external dependencies.

### CODE REVIEW

File: devshell.nix [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-  env = [
-    {
-      name = "DEVSHELL_NO_MOTD";
-      value = 1;
-    }
-  ];
-
-  commands = [
-    { package = perSystem.nixos-facter.default; }
-    { package = pkgs.nix-unit; }
```
Comment (Mic92) [inline devshell.nix:12]: nix-unit doesn't seem to be used by the unit-test.

File: flake.lock [hunk 2]
```diff
@@ -25,123 +23,13 @@
...
-        "rev": "b1d9ab70662946ef0850d488da1c9019f3a9752a",
-        "type": "github"
-      },
-      "original": {
-        "owner": "numtide",
-        "repo": "flake-utils",
-        "type": "github"
-      }
-    },
-    "gomod2nix": {
```
Comment (Mic92) [inline flake.lock:89]: Problem is that it's not possible to override dependencies like this.

File: devshell.nix (hunk 1 above)
Comment (brianmcgee) [reply devshell.nix:12]: `nix-unit` is being used by the `unit-tests` package:

https://github.com/numtide/nixos-facter-modules/blob/0211209c873004f9ee3dc5a24de713f2737c5cd0/packages/unit-tests.nix#L8-L14

It was in the devshell to allow for running it manually / targeting specific tests etc.

Comment (Mic92) [reply devshell.nix:12]: I see.

File: devshell.nix [hunk 3]
```diff
@@ -1,14 +1 @@
...
-  ];
-}
+{ pkgs, ... }: pkgs.mkShell { packages = [ pkgs.nix-unit ]; }
```
Comment (Mic92) [inline devshell.nix:1]:
```suggestion
{ pkgs, ... }: pkgs.mkShellNoCC { packages = [ pkgs.nix-unit ]; }
```

**Review by brianmcgee — APPROVED** [review]

**Pull request merged**